simulator/src/main/java/simblock/node/Farmer.java

```diff
@@ -0,0 +1,101 @@
+package simblock.node;
+
+import simblock.block.Block;
+import simblock.block.ChiaBlock;
+import simblock.simulator.Simulator;
+import simblock.simulator.Timer;
+import simblock.task.AbstractMintingTask;
+
+import static simblock.settings.SimulationConfiguration.CHIA_K;
+import static simblock.settings.SimulationConfiguration.NUM_OF_NODES;
+import static simblock.simulator.Simulator.arriveBlock;
+
+public class Farmer extends Node {
+
+    // this is used to specify the size of the PoSpaceCount array and corresponds to how many levels/depths we keep track of
+    private static final int DEPTH = NUM_OF_NODES * 2;
+
+    // how many blocks this farmer has mined at a particular level
+    private final int[] PoSpaceCount;
+
+    // block that is going to be minted/farmed off of
+    private ChiaBlock mintingBlock;
```

Comment:
Node uses currentBlock to both keep track of the best path seen so far and to determine which block to mine off of. Since Chia mines off the first three blocks received, and not off the best path, we use a separate attribute to keep track of which block to mine off of